frontend/src/components/Playback/MusicVisualizer.vue

```diff
@@ -0,0 +1,44 @@
+<template>
+  <div ref="musicVisualizer" />
+</template>
+
+<script setup lang="ts">
+import { ref, onMounted, onBeforeUnmount } from 'vue';
+import AudioMotionAnalyzer from 'audiomotion-analyzer';
+import { mediaElementRef } from '@/store';
+
+let visualizer: AudioMotionAnalyzer;
+const musicVisualizer = ref(undefined);
+
+onMounted(() => {
+  visualizer = new AudioMotionAnalyzer(musicVisualizer.value, {
+    source: mediaElementRef.value,
+    mode: 2,
+    gradient: 'prism',
+    reflexRatio: 0.025,
+    overlay: true,
+    showBgColor: true,
+    bgAlpha: 0,
+    fftSize: 16_384,
+    frequencyScale: 'bark',
+    showScaleX: false,
+    smoothing: 0.9
+  });
+});
+
+onBeforeUnmount(() => {
+  if (visualizer) {
+    visualizer.toggleAnalyzer();
+    visualizer.disconnectInput();
+    visualizer.disconnectOutput();
+
+    if (mediaElementRef.value) {
+      const audioCtx = new AudioContext();
+
+      audioCtx
+        .createMediaElementSource(mediaElementRef.value)
+        .connect(audioCtx.destination);
+    }
```

Comment:
I guess we need this at mount to connect to the visualizer instead of onBeforeUnmount? Why?

Reply:
When you instantiate the visualizer it [looks like](https://github.com/hvianna/audioMotion-analyzer/blob/de9fb304295208535f434c9c30a3b66ae6d4608f/src/audioMotion-analyzer.js#L152C2-L166) it inserts itself between the audio source and the speakers automatically (which is why it's not in onMount), so when you disconnect the inputs and outputs it also ends up disconnecting the speakers. So I added those lines to reconnect the audio source directly to the speakers when you disable the visualizer.